win32ss/gdi/ntgdi/dibobj.c

```diff
@@ -1279,110 +1320,213 @@ NtGdiStretchDIBitsInternal(
         }
         _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
         {
-            ExFreePoolWithTag(pvBits, 'pmeT');
+            ExFreePoolWithTag(pvBits, TAG_DIB);
             _SEH2_YIELD(return 0);
         }
-        _SEH2_END
+        _SEH2_END;
     }
     else
     {
         pvBits = NULL;
     }
 
-    /* FIXME: Locking twice is cheesy, coord tranlation in UM will fix it */
-    if (!(pdc = DC_LockDc(hdc)))
+    /* Here we select between the dwRop with SRCCOPY or not. */
+    if (dwRop == SRCCOPY)
```

Comment:
I actually wonder if all this stuff has any benefit over the other path. Anyway, it it works, I'm fine with it. We can optimize it later.

Reply:
I can tell you that you need both of these paths. The first one handles items like the Sims graphics and the other path is necessary to satisfy the gdi32:bitmap tests or you will get more failure when (dwRop != SRCCOPY). If this was better understood, it might be possible to fix this in the first section of the code though.